packages/melos/lib/src/commands/clean.dart

```diff
@@ -40,7 +40,19 @@ mixin _CleanMixin on _Melos {
     for (final generatedPubFilePath in pathsToClean) {
       final file = File(join(package.path, generatedPubFilePath));
       if (file.existsSync()) {
-        file.deleteSync(recursive: true);
+        await file.delete(recursive: true);
+      }
+    }
+
+    // Remove any Melos generated dependency overrides from
+    // `pubspec_overrides.yaml`.
+    final pubspecOverridesFile =
+        File(join(package.path, 'pubspec_overrides.yaml'));
+    if (pubspecOverridesFile.existsSync()) {
+      final contents = await pubspecOverridesFile.readAsString();
+      final updatedContents = mergeMelosPubspecOverrides({}, contents);
+      if (updatedContents != null) {
+        await pubspecOverridesFile.writeAsString(updatedContents);
       }
```

Comment:
I'm not sure but do we want to delete the file here in an `else` since it's empty and since it was most likely created by Melos?

Reply:
`mergeMelosPubspecOverrides` allows overrides that don't come from melos, so passing in `{}` does not guarantee that `pubspec_overrides.yaml` is empty. But we can check for that and if there are no other overrides, then I think it's a good idea to delete the file.

Reply:
Done